src/uu/csplit/src/csplit.rs

```diff
@@ -704,43 +710,83 @@ mod tests {
+                .long(options::SUFFIX_FORMAT)
+                .value_name("FORMAT")
+                .help("use sprintf FORMAT instead of %02d"),
+        )
+        .arg(
+            Arg::with_name(options::PREFIX)
+                .short("f")
+                .long(options::PREFIX)
+                .value_name("PREFIX")
+                .help("use PREFIX instead of 'xx'"),
+        )
+        .arg(
+            Arg::with_name(options::KEEP_FILES)
+                .short("k")
+                .long(options::KEEP_FILES)
+                .help("do not remove output files on errors"),
+        )
+        .arg(
+            Arg::with_name(options::SUPPRESS_MATCHED)
+                .long(options::SUPPRESS_MATCHED)
+                .help("suppress the lines matching PATTERN"),
+        )
+        .arg(
+            Arg::with_name(options::DIGITS)
+                .short("n")
+                .long(options::DIGITS)
+                .value_name("DIGITS")
+                .help("use specified number of digits instead of 2"),
+        )
+        .arg(
+            Arg::with_name(options::SILENT)
+                .short("s")
+                .long(options::SILENT)
+                .help("do not print counts of output file sizes"),
         )
-        .optopt("f", PREFIX_OPT, "use PREFIX instead of 'xx'", "PREFIX")
-        .optflag("k", KEEP_FILES_OPT, "do not remove output files on errors")
-        .optflag(
-            "",
-            SUPPRESS_MATCHED_OPT,
-            "suppress the lines matching PATTERN",
+        .arg(
+            Arg::with_name(options::QUIET)
+                .long(options::QUIET)
+                .help("same as --silent"),
         )
-        .optopt(
-            "n",
-            DIGITS_OPT,
-            "use specified number of digits instead of 2",
-            "DIGITS",
+        .arg(
+            Arg::with_name(options::ELIDE_EMPTY_FILES)
+                .short("z")
+                .long(options::ELIDE_EMPTY_FILES)
+                .help("remove empty output files"),
         )
-        .optflag("s", QUIET_OPT, "do not print counts of output file sizes")
-        .optflag("z", ELIDE_EMPTY_FILES_OPT, "remove empty output files")
-        .parse(args);
+        .arg(Arg::with_name(options::FILE).hidden(true))
+        .arg(Arg::with_name(options::PATTERN).hidden(true).multiple(true))
+        .after_help(LONG_HELP)
+        .get_matches_from(args);
 
     // check for mandatory arguments
-    if matches.free.is_empty() {
+    let file_value = matches.value_of(options::FILE);
+    if file_value.is_none() {
         show_error!("missing operand");
         exit!(1);
     }
-    if matches.free.len() == 1 {
-        show_error!("missing operand after '{}'", matches.free[0]);
+
+    // get the file to split
+    let file_name = file_value.unwrap();
+
+    let patterns_values = matches.values_of(options::PATTERN);
+    if patterns_values.is_none() {
+        show_error!("missing operand after '{}'", file_name);
```

Comment:
clap should take care of these cases if you tell it which `.arg` are `required`, `required_unless`, etc.